xbmc/pvr/PVRActionListener.cpp

```diff
@@ -124,60 +134,66 @@ bool CPVRActionListener::OnAction(const CAction &action)
-      {
-        CServiceBroker::GetPVRManager().ChannelPreviewUp(ChannelSwitchMode::NO_SWITCH);
-      }
-      else if (CServiceBroker::GetSettings().GetInt(CSettings::SETTING_PVRPLAYBACK_CHANNELENTRYTIMEOUT) > 0)
-      {
-        CServiceBroker::GetPVRManager().ChannelPreviewUp(ChannelSwitchMode::DELAYED_SWITCH);
-      }
-      else
-      {
-        const CPVRChannelPtr currentChannel(CServiceBroker::GetPVRManager().GetCurrentChannel());
-        const CFileItemPtr item(CServiceBroker::GetPVRManager().ChannelGroups()->Get(currentChannel->IsRadio())->GetSelectedGroup()->GetByChannelUp(currentChannel));
-        CServiceBroker::GetPVRManager().GUIActions()->SwitchToChannel(item, false);
-      }
-      
+      CServiceBroker::GetPVRManager().GUIActions()->GetChannelNavigator().SelectNextChannel(GetChannelSwitchMode(action.GetID()));
       return true;
     }
+
     case ACTION_MOVE_DOWN:
     case ACTION_PREV_ITEM:
     case ACTION_CHANNEL_DOWN:
     {
       if (!bIsPlayingPVR)
         return false;
 
-      if (action.GetID() == ACTION_MOVE_DOWN && // only up/down shows only a preview, all others do switch
-          CServiceBroker::GetSettings().GetBool(CSettings::SETTING_PVRPLAYBACK_CONFIRMCHANNELSWITCH))
-      {
-        CServiceBroker::GetPVRManager().ChannelPreviewDown(ChannelSwitchMode::NO_SWITCH);
-      }
-      else if (CServiceBroker::GetSettings().GetInt(CSettings::SETTING_PVRPLAYBACK_CHANNELENTRYTIMEOUT) > 0)
-      {
-        CServiceBroker::GetPVRManager().ChannelPreviewDown(ChannelSwitchMode::DELAYED_SWITCH);
-      }
-      else
+      CServiceBroker::GetPVRManager().GUIActions()->GetChannelNavigator().SelectPreviousChannel(GetChannelSwitchMode(action.GetID()));
+      return true;
+    }
+
+    case ACTION_SELECT_ITEM:
+    {
+      if (!bIsPlayingPVR)
+        return false;
+
+      if (g_windowManager.IsWindowActive(WINDOW_FULLSCREEN_VIDEO) || g_windowManager.IsWindowActive(WINDOW_VISUALISATION))
       {
-        const CPVRChannelPtr currentChannel(CServiceBroker::GetPVRManager().GetCurrentChannel());
-        const CFileItemPtr item(CServiceBroker::GetPVRManager().ChannelGroups()->Get(currentChannel->IsRadio())->GetSelectedGroup()->GetByChannelDown(currentChannel));
-        CServiceBroker::GetPVRManager().GUIActions()->SwitchToChannel(item, false);
-      }
+        if (CServiceBroker::GetSettings().GetBool(CSettings::SETTING_PVRPLAYBACK_CONFIRMCHANNELSWITCH) &&
+            CServiceBroker::GetPVRManager().GUIActions()->GetChannelNavigator().IsPreview())
+        {
+          // switch to previewed channel
+          CServiceBroker::GetPVRManager().GUIActions()->GetChannelNavigator().SwitchToCurrentChannel();
+          return true;
+        }
+        else
+        {
+          // activate player controls
+          CGUIDialog *pOSD = g_windowManager.GetDialog(WINDOW_DIALOG_VIDEO_OSD);
+          if (pOSD && !pOSD->IsDialogRunning())
+            pOSD->Open();
```

Comment:
BTW do be more clear. You can't map buttons to the player info because it's not a dialog or window. This info is only handled by an info bool `PLAYER_SHOWINFO`. Therefore that only way to get this working is asking for the global select action if we want the user to customize the buttons for the fullscreen/visualization windows.

If player info would be a dialog, this would be possible, but in that case STOP, PAUSE etc. should be delegated to the underlying fullscreen window if not handled.